modules/core/arrow-data/src/test/kotlin/arrow/data/MooreTest.kt

```diff
@@ -0,0 +1,90 @@
+package arrow.data
+
+import arrow.core.Id
+import arrow.instances.ForMoore
+import arrow.test.UnitSpec
+import arrow.test.laws.ComonadLaws
+import arrow.test.laws.FunctorLaws
+import arrow.typeclasses.Eq
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldBe
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class MooreTest : UnitSpec() {
+
+  init {
+
+    fun handle(x: Int): Moore<Int, Int> = Moore(x, ::handle)
+    val intMoore: (Int) -> MooreOf<Int, Int> = { x: Int -> Moore(x, ::handle) }
+
+    val EQ = object : Eq<MooreOf<Int, Int>> {
+      override fun MooreOf<Int, Int>.eqv(b: MooreOf<Int, Int>): Boolean =
+          this.fix().extract() == b.fix().extract()
+    }
+
+    ForMoore<Int>() extensions {
+      testLaws(
+          FunctorLaws.laws(Moore.functor(), intMoore, EQ),
+          ComonadLaws.laws(Moore.comonad(), intMoore, EQ)
```

Comment:
Doesn't Comonad imply Functor? You may not need to test both.

Reply:
You are right 👌